src/routes/docs/tutorials/pagination/step-8/+page.markdoc

```diff
@@ -0,0 +1,275 @@
+---
+layout: tutorial
+title: Implementing the cursor pagination
+description: Explore cursor pagination and how the React app accesses the appwrite backend using this method.
+step: 8
+---
+In this step, we'll delve into adding the cursor pagination method to our app.
+
+# What is cursor pagination? {% #what-is-cursor-pagination %}
+Cursor pagination uses a unique identifier(often a timestamp, ID, or another consistently ordered field) for a specific document. This identifer acts as a pointer to the next document we want to start querying from to get the next page of data. For instance, if you have a dataset of blog posts ordered by their creation date and you want to retrieve post after a specific data, you would use that date as the cursor.
+
+With Appwrite, cursor pagination is achieved using the `Query.cursorAfter(lastId)` or the `Query.cursorBefore(firstId)` query methods. Here is an example using Appwrite's `listDocuments()` function to retrieve the next page of documents:
+
+```js
+const page2 = await databases.listDocuments(
+    '[DATABASE_ID]',
+    '[COLLECTION_ID]',
+    [
+        Query.limit(25),
+        Query.cursorAfter(lastId),
+    ]
+);
+```
+Cursor pagination is typically used in scenarios where the dataset is large, and the exact number of total items might not be known upfront. Here are different ways you can utilize cursor pagination:
+
+# Creating the Pagination Component {% #creating-the-pagination-component %}
+{% tabs %}
+{% tabsitem #Basic-cursor-pagination title="Basic Cursor Pagination" %}
+
+Create a file and named `src/components/BasicCursorPagination.jsx`, insert the following code:
+```js
+
+function CursorPagination(props) {
+  const {
+    hasNext,
+    loadMore,
+    loadPrev,
+    currentPage
+  } = props;
+
+  return (
+    <div>
+      <button disabled={currentPage === 1} onClick={loadPrev}>
+        Load Previous
+      </button>
+      <span>Page {currentPage}</span>
+      <button disabled={!hasNext} onClick={loadMore}>
+        Load More
+      </button>
+    </div>
+  );
+}
+
+export default CursorPagination;
+
+```
+This is the most straightforward use of cursor pagination. You have "Load Previous" and "Load More" buttons, and you fetch data based on the last or first items's cursor.
+{% /tabsitem %}
+{% tabsitem #infinite-scroll-with-cursor title="Infinite Scroll with Cursor" %}
+Create a file and named `src/components/InfiniteScroll.jsx`, insert the following code:
+```js
+import React, { useEffect } from 'react';
+
+function InfiniteScroll(props) {
```

Comment:
Don't think this is working haha~ 

Let's just use a button to load more 😅

Reply:
I guess todos were too little for the required window. This may be a tutorial on infinite scrolling image gallery searches or something that would make it more practical to use with.

Reply:
Yeah it's fine. They can go figure out the UI, not within scope.